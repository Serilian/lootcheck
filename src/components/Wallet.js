import React, {Component} from 'react';
import {connect} from 'react-redux';

export class Wallet extends Component {
    constructor(props){
        super(props);
    }

    render() {
        const {balance} = this.props;

        return (
            <div>
                <h2>Wallet</h2>
                <h4 className={'balance'}>Balance is: {balance}</h4>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    balance: state.balance
});

export default connect(mapStateToProps)(Wallet);