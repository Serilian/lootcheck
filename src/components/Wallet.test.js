import React from 'react';
import {shallow} from 'enzyme';
import {Wallet} from './Wallet';

describe("Given a wallet component", ()=>{
    let wallet;
    let MOCK_BALANCE = 20;

    beforeEach(()=>{
     const props = {balance: MOCK_BALANCE};
      wallet  = shallow(<Wallet {...props}/>);
    });

    it('Then it renders correctly', ()=>{
        expect(wallet.debug()).toMatchSnapshot();
    });
    it('Then displays the balance from props', ()=>{
        expect(wallet.find('.balance').text()).toEqual(`Balance is: ${MOCK_BALANCE}`)
    });

});