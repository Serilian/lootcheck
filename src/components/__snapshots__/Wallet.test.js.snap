// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a wallet component Then it renders correctly 1`] = `
"<div>
  <h2>
    Wallet
  </h2>
  <h4 className=\\"balance\\">
    Balance is: 
    20
  </h4>
</div>"
`;
